src/dialect/mod.rs

```diff
@@ -457,88 +461,57 @@ pub trait Dialect: Debug + Any {
         None
     }
 
-    // The following precedence values are used directly by `Parse` or in dialects,
-    // so have to be made public by the dialect.
-
-    /// Return the precedence of the `::` operator.
+    /// Decide the lexical Precedence of operators.
     ///
-    /// Default is 50.
-    fn prec_double_colon(&self) -> u8 {
-        DOUBLE_COLON_PREC
-    }
-
-    /// Return the precedence of `*`, `/`, and `%` operators.
-    ///
-    /// Default is 40.
-    fn prec_mul_div_mod_op(&self) -> u8 {
-        MUL_DIV_MOD_OP_PREC
-    }
-
-    /// Return the precedence of the `+` and `-` operators.
-    ///
-    /// Default is 30.
-    fn prec_plus_minus(&self) -> u8 {
-        PLUS_MINUS_PREC
-    }
-
-    /// Return the precedence of the `BETWEEN` operator.
-    ///
-    /// For example `BETWEEN <low> AND <high>`
-    ///
-    /// Default is 22.
-    fn prec_between(&self) -> u8 {
-        BETWEEN_PREC
-    }
-
-    /// Return the precedence of the `LIKE` operator.
-    ///
-    /// Default is 19.
-    fn prec_like(&self) -> u8 {
-        LIKE_PREC
-    }
-
-    /// Return the precedence of the unary `NOT` operator.
-    ///
-    /// For example `NOT (a OR b)`
-    ///
-    /// Default is 15.
-    fn prec_unary_not(&self) -> u8 {
-        UNARY_NOT_PREC
+    /// Uses (APPROXIMATELY) <https://www.postgresql.org/docs/7.0/operators.htm#AEN2026> as a reference
+    fn prec_value(&self, prec: Precedence) -> u8 {
```

Comment:
I think this is a good idea. 

My only potential concern with this approach is that it could be potentially slower, as it now has many more function calls, however I don't actually think we have a sqlparser benchmark so at this point so trying to make such a change would likely be a pre-mature optimization. 

If anyone needs additional performance, we can do that as follow on PRs